tests/run-test

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+#
+# Extract the git bundle, prepare the git repository, and run a single test.
+# This is intended to be run from the Meson test framework.
+#
+set -e
+
+fail() {
+	echo "$1"
+	exit 1
+}
+
+TEST="$1"
+
+if [ -z "$1" ]; then fail "usage: run-test <test>"; fi
+if [ -z "${SOURCE_ROOT}" ]; then fail "SOURCE_ROOT not set"; fi
+if [ -z "${TEST_DIR}" ]; then fail "TEST_DIR not set"; fi
+if [ -z "${BUILD_DIR}" ]; then fail "BUILD_DIR not set"; fi
+if [ ! -x "${SOURCE_ROOT}/umpf" ]; then fail "umpf not found in SOURCE_ROOT"; fi
+if [ -z "$(command -v git)" ]; then fail "git not found in PATH"; fi
+
+echo "---------------------------------------- preparing git repository"
+rm -rf "${BUILD_DIR:?}/${TEST}"
```

Comment:
You're already checking if `BUILD_DIR` is defined in line 18, so no need for `:?` here.

Reply:
ok